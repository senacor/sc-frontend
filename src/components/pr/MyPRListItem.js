import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import moment from 'moment/moment';
import styles from './MyPRListStyle';

export class MyPRListItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      reviewer: props.reviewer,
      deadline: props.deadline,
      occasion: props.occasion,
      status: props.status
    };
  }

  render() {
    const { classes } = this.props;
    const { reviewer, deadline, occasion, status } = this.state;
    return (
      <ListItem className={classes.listItem}>
        <ListItemText>
          <div className={classes.divItemText}>
            <Typography className={classes.typography}>
              Beurteiler: {reviewer}
            </Typography>
            {deadline ? (
              <Typography className={classes.typography}>
                Datum: {moment(deadline).format('DD.MM.YY')}
              </Typography>
            ) : (
              ''
            )}

            <Typography className={classes.typography}>{occasion}</Typography>
            <Typography
              className={
                status === 'DONE'
                  ? classes.typographyDone
                  : classes.typographyGreen
              }
            >
              {status}
            </Typography>
          </div>
        </ListItemText>
      </ListItem>
    );
  }
}

export default withStyles(styles)(MyPRListItem);
