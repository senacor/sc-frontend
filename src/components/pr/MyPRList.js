import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../../actions/index';
import withLoading from '../hoc/Loading';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Icon from '@material-ui/core/Icon';
import Hidden from '@material-ui/core/Hidden';
import AddIcon from '@material-ui/icons/Add';
import PrState from './PrState';
import PrSalary from './PrSalary';
import PrEmployment from './PrEmployment';
import PrSheet from './PrSheet';
import Card from '@material-ui/core/Card';
import moment from 'moment';
import { Link } from 'react-router-dom';
import List from '@material-ui/core/List';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import styles from './MyPRListStyle';
import { default as MyPRListItem } from './MyPRListItem';

export class MyPRList extends React.Component {
  constructor(props) {
    super(props);
    let prOpen = props.prs.length !== 0 ? props.prs[0] : null;
    this.state = {
      prs: props.prs,
      prOpen: prOpen,
      filters: props.filters
        ? props.filters
        : {
            reviewer: 'ALL',
            occasion: 'ALL'
          },
      sortOrder: 'asc'
    };
  }

  translateStatus = status => {
    switch (status) {
      case 'PREPARATION':
        return 'In Vorbereitung';
      case 'EXECUTION':
        return 'In Durchführung';
      case 'POST_PROCESSING':
        return 'Nachbearbeitung';
      case 'DONE':
        return 'Fertig';
      default:
        return 'In Vorbereitung';
    }
  };

  translateOccasion = occasion => {
    switch (occasion) {
      case 'ON_DEMAND':
        return 'Auf Nachfrage';
      case 'YEARLY':
        return 'jährlich';
      case 'QUARTERLY':
        return 'vierteljährlich';
      case 'END_PROBATION':
        return 'Ende der Probezeit';
      default:
        return 'Auf Nachfrage';
    }
  };

  handleClick = () => {
    this.props.addPr();
  };

  openAnotherSheet = pr => {
    this.setState({
      prOpen: pr
    });
  };

  switchDateOrder = () => {
    this.setState({
      sortOrder: this.state.sortOrder === 'asc' ? 'desc' : 'asc'
    });
    this.props.ChangePrSortOrderToProvidedDirection(this.state.sortOrder);
  };

  filterPR = pr => {
    const { filters } = this.state;
    return (
      //pr.employee.firstName === 'Lionel' &&
      (filters.reviewer === 'ALL' || pr.supervisor === filters.reviewer) &&
      (filters.occasion === 'ALL' || pr.occasion === filters.occasion)
    );
  };

  handleFilter = event => {
    const oldfilter = this.state.filters;
    switch (event.target.name) {
      case 'Reviewer':
        this.setState({
          filters: {
            ...oldfilter,
            reviewer: event.target.value
          }
        });
        break;
      case 'Occasion':
        this.setState({
          filters: {
            ...oldfilter,
            occasion: event.target.value
          }
        });
        break;
      default:
        break;
    }
  };

  componentDidMount() {
    this.props.ChangePrSortOrderToProvidedDirection(this.state.sortOrder);
  }

  render() {
    const { classes, prs } = this.props;
    const { prOpen, sortOrder } = this.state;
    return (
      <div>
        <Typography variant="display1" paragraph>
          Performance Reviews
        </Typography>
        <Button
          id="sortButton"
          className={classes.button}
          variant="outlined"
          onClick={this.switchDateOrder}
        >
          Datum
          <Icon className={classes.rightIcon}>
            {sortOrder === 'asc' ? 'arrow_upward' : 'arrow_downward'}
          </Icon>
        </Button>

        <Select
          id="_SelectIdFilter"
          value={this.state.filters.reviewer}
          onChange={this.handleFilter}
          displayEmpty
          name="Reviewer"
          className={classes.buttonDesktop}
        >
          <MenuItem value="ALL">
            <div className={classes.filterDesktop}>Alle</div>
          </MenuItem>
          {Array.from(new Set(prs.map(pr => pr.supervisor))).map(reviewer => {
            return (
              <MenuItem key={reviewer} value={reviewer}>
                <div className={classes.filterDesktop}>{reviewer}</div>
              </MenuItem>
            );
          })}
        </Select>
        <Select
          id="_SelectIdFilter"
          value={this.state.filters.occasion}
          onChange={this.handleFilter}
          displayEmpty
          name="Occasion"
          className={classes.buttonDesktop}
        >
          <MenuItem value="ALL">
            <div className={classes.filterDesktop}>Alle</div>
          </MenuItem>
          {['ON_DEMAND', 'YEARLY', 'QUARTERLY', 'END_PROBATION'].map(
            occasion => {
              return (
                <MenuItem key={occasion} value={occasion}>
                  <div className={classes.filterDesktop}>
                    {this.translateOccasion(occasion)}
                  </div>
                </MenuItem>
              );
            }
          )}
        </Select>

        <Hidden smDown>
          <Button
            className={classes.buttonDesktop}
            variant="raised"
            onClick={this.handleClick}
          >
            PR beantragen
            <Icon className={classes.rightIcon}>add</Icon>
          </Button>
        </Hidden>

        <Hidden smDown>
          <div className={classes.container}>
            <div className={classes.root}>
              {prs.filter(this.filterPR).map(pr => {
                return (
                  <Button
                    key={pr.id}
                    id={pr.id}
                    className={classes.buttonList}
                    onClick={() => {
                      this.openAnotherSheet(pr);
                    }}
                  >
                    <List
                      className={
                        prOpen === pr ? classes.openList : classes.list
                      }
                    >
                      <MyPRListItem
                        reviewer={pr.supervisor}
                        deadline={pr.deadline}
                        occasion={this.translateOccasion(pr.occasion)}
                        status={this.translateStatus(pr.status)}
                      />
                    </List>
                  </Button>
                );
              })}
            </div>
            {prs.length === 0 ? (
              ''
            ) : (
              <div className={classes.cardContainerRow}>
                <Card className={classes.cardColumnSheet}>
                  <Typography variant="body2" className={classes.title}>
                    SHEET
                  </Typography>
                  <PrSheet prById={prs[0]} />
                </Card>
                <div className={classes.cardContainerColumn}>
                  <Card className={classes.cardColumn}>
                    <Typography variant="body2" className={classes.title}>
                      GEHALT UND ANSTELLUNG
                    </Typography>
                    <PrSalary prById={prs[0]} />
                    <PrEmployment prById={prs[0]} />
                  </Card>

                  <Card className={classes.cardColumn}>
                    <Typography variant="body2" className={classes.title}>
                      STATUS
                    </Typography>
                    <PrState expanded={this.state.expanded} />
                  </Card>
                </div>
              </div>
            )}
          </div>
        </Hidden>
        <Hidden smUp>
          {prs.filter(this.filterPR).map(pr => {
            return (
              <Link
                key={pr.id}
                to={`/prs/${pr.id}`}
                style={{ textDecoration: 'none' }}
              >
                <List className={classes.list}>
                  <MyPRListItem
                    reviewer={pr.supervisor}
                    deadline={pr.deadline}
                    occasion={this.translateOccasion(pr.occasion)}
                    status={this.translateStatus(pr.status)}
                  />
                </List>
              </Link>
            );
          })}
        </Hidden>
        <Hidden smUp>
          <Button
            variant="fab"
            color="primary"
            aria-label="add"
            className={classes.buttonMobile}
            onClick={this.handleClick}
          >
            <AddIcon className={classes.icon} />
          </Button>
        </Hidden>
      </div>
    );
  }
}

export const StyledComponent = withStyles(styles)(MyPRList);
export default connect(
  state => ({
    prs: state.prs.prsList,
    isLoading: state.isLoading
  }),
  {
    fetchPrs: actions.fetchPrs,
    addPr: actions.addPr,
    addSupervisor: actions.addSupervisor,
    ChangePrSortOrderToProvidedDirection:
      actions.ChangePrSortOrderToProvidedDirection
  }
)(withLoading(props => props.fetchPrs())(StyledComponent));
