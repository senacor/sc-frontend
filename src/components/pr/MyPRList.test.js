import React from 'react';
import { StyledComponent } from './MyPRList';
import { createShallow } from '@material-ui/core/test-utils';
import MyPRListItem from './MyPRListItem';

describe('MyPRList Component', () => {
  let shallow = createShallow({ dive: true });

  const prs = [
    {
      id: 1,
      employee: {
        id: 1,
        login: 'lschäfer',
        firstName: 'Lionel',
        lastName: 'Schäfer',
        title: 'DR',
        email: 'lionel.schäfer@senacor.com',
        entryDate: '2004-05-10',
        salaries: [
          {
            workingHoursPerWeek: 40,
            ote: 45000,
            fte: 1,
            basicSalary: 40500,
            validFrom: '2004-05-10'
          },
          {
            workingHoursPerWeek: 40,
            ote: 50000,
            fte: 1,
            basicSalary: 40000,
            validFrom: '2005-05-10'
          }
        ],
        employment: {
          endOfProbationPeriod: '2004-05-10',
          jobGradings: [
            {
              positionId: 1,
              salaryLevel: 1,
              validFrom: '2004-05-09',
              competence: 'DEVELOPMENT'
            },
            {
              positionId: 2,
              salaryLevel: 2,
              validFrom: '2005-05-09',
              competence: 'DEVELOPMENT'
            }
          ],
          leaves: {
            unpaidLeave: [
              {
                from: '2004-05-10',
                to: '2005-03-31'
              }
            ],
            maternityLeave: [
              {
                from: '2004-05-10',
                to: '2004-10-11'
              }
            ],
            parentalLeave: [
              {
                from: '2004-05-10',
                to: '2004-08-28'
              }
            ],
            sabbatical: [
              {
                from: '2004-05-10',
                to: '2004-06-07'
              },
              {
                from: '2004-05-10',
                to: '2005-02-10'
              },
              {
                from: '2004-05-10',
                to: '2005-03-26'
              }
            ]
          }
        }
      },
      supervisor: 'ttran',
      occasion: 'YEARLY',
      status: 'PREPARATION',
      deadline: '2015-05-11',
      _links: {
        self: {
          href: 'http://localhost:8010/api/v1/prs/1'
        }
      }
    },
    {
      id: 2,
      employee: {
        id: 1,
        login: 'lschäfer',
        firstName: 'Lionel',
        lastName: 'Schäfer',
        title: 'DR',
        email: 'lionel.schäfer@senacor.com',
        entryDate: '2004-05-10',
        salaries: [
          {
            workingHoursPerWeek: 40,
            ote: 45000,
            fte: 1,
            basicSalary: 40500,
            validFrom: '2004-05-10'
          },
          {
            workingHoursPerWeek: 40,
            ote: 50000,
            fte: 1,
            basicSalary: 40000,
            validFrom: '2005-05-10'
          }
        ],
        employment: {
          endOfProbationPeriod: '2004-05-10',
          jobGradings: [
            {
              positionId: 1,
              salaryLevel: 1,
              validFrom: '2004-05-09',
              competence: 'DEVELOPMENT'
            },
            {
              positionId: 2,
              salaryLevel: 2,
              validFrom: '2005-05-09',
              competence: 'DEVELOPMENT'
            }
          ],
          leaves: {
            unpaidLeave: [
              {
                from: '2004-05-10',
                to: '2005-03-31'
              }
            ],
            maternityLeave: [
              {
                from: '2004-05-10',
                to: '2004-10-11'
              }
            ],
            parentalLeave: [
              {
                from: '2004-05-10',
                to: '2004-08-28'
              }
            ],
            sabbatical: [
              {
                from: '2004-05-10',
                to: '2004-06-07'
              },
              {
                from: '2004-05-10',
                to: '2005-02-10'
              },
              {
                from: '2004-05-10',
                to: '2005-03-26'
              }
            ]
          }
        }
      },
      supervisor: 'test',
      occasion: 'ON_DEMAND',
      status: 'PREPARATION',
      deadline: '2222-05-11',
      _links: {
        self: {
          href: 'http://localhost:8010/api/v1/prs/1'
        }
      }
    }
  ];

  it('displays the MyPRList', () => {
    const component = shallow(<StyledComponent prs={prs} />);

    expect(component).toMatchSnapshot();
  });

  it('filters PRs according to a reviewer', () => {
    const wrapper = shallow(<StyledComponent prs={prs} />);
    expect(wrapper.find('[reviewer="ttran"]')).toHaveLength(2);

    wrapper.setState({ filters: { reviewer: 'S1', occasion: 'ALL' } });

    expect(wrapper.find('[reviewer="ttran"]')).toHaveLength(0);
  });

  it('should filter by occasion', () => {
    const wrapper = shallow(
      <StyledComponent
        prs={prs}
        filters={{ reviewer: 'ALL', occasion: 'ALL' }}
      />
    );
    const text_ondemand = wrapper.instance().translateOccasion('ON_DEMAND');
    const text_yearly = wrapper.instance().translateOccasion('YEARLY');
    expect(wrapper.find('[occasion="' + text_ondemand + '"]')).toHaveLength(2);
    expect(wrapper.find('[occasion="' + text_yearly + '"]')).toHaveLength(2);

    wrapper.setState({ filters: { reviewer: 'ALL', occasion: 'ON_DEMAND' } });

    expect(wrapper.find('[occasion="' + text_ondemand + '"]')).toHaveLength(2);
    expect(wrapper.find('[occasion="' + text_yearly + '"]')).toHaveLength(0);
  });

  it('should filter by reviewer and occasion at the same time', () => {
    const wrapper = shallow(
      <StyledComponent
        prs={prs}
        filters={{ reviewer: 'ALL', occasion: 'ALL' }}
      />
    );
    const text_ondemand = wrapper.instance().translateOccasion('ON_DEMAND');
    const text_yearly = wrapper.instance().translateOccasion('YEARLY');
    expect(wrapper.find('[reviewer="ttran"]')).toHaveLength(2);
    expect(wrapper.find('[reviewer="test"]')).toHaveLength(2);
    expect(wrapper.find('[occasion="' + text_ondemand + '"]')).toHaveLength(2);
    expect(wrapper.find('[occasion="' + text_yearly + '"]')).toHaveLength(2);

    wrapper.setState({ filters: { reviewer: 'ttran', occasion: 'ON_DEMAND' } });

    expect(wrapper.find('[reviewer="ttran"]')).toHaveLength(0);
  });

  it('sorts the MyPRList in given order', () => {
    const wrapper = shallow(<StyledComponent prs={prs} />);
    expect(wrapper.find('[deadline]').get(0).props.deadline).toEqual(
      '2015-05-11'
    );
    expect(wrapper.find('[deadline]').get(1).props.deadline).toEqual(
      '2222-05-11'
    );
    wrapper.setState({ sortDateInAscOrder: false });
    expect(wrapper.find('[deadline]').get(0).props.deadline).toEqual(
      '2222-05-11'
    );
    expect(wrapper.find('[deadline]').get(1).props.deadline).toEqual(
      '2015-05-11'
    );
  });
});
