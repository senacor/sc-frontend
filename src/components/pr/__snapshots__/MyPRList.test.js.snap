// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPRList Component displays the MyPRList 1`] = `
<div>
  <WithStyles(Typography)
    paragraph={true}
    variant="display1"
  >
    Performance Reviews
  </WithStyles(Typography)>
  <WithStyles(Button)
    id="sortButton"
    onClick={[Function]}
    variant="outlined"
  >
    Datum
    <WithStyles(Icon)>
      arrow_upward
    </WithStyles(Icon)>
  </WithStyles(Button)>
  <WithStyles(Select)
    className="MyPRList-buttonDesktop-5"
    displayEmpty={true}
    name="Reviewer"
    onChange={[Function]}
    value="ALL"
  >
    <WithStyles(MenuItem)
      value="ALL"
    >
      <div
        className="MyPRList-filterDesktop-6"
      >
        Alle
      </div>
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="ttran"
      value="ttran"
    >
      <div
        className="MyPRList-filterDesktop-6"
      >
        ttran
      </div>
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="test"
      value="test"
    >
      <div
        className="MyPRList-filterDesktop-6"
      >
        test
      </div>
    </WithStyles(MenuItem)>
  </WithStyles(Select)>
  <WithStyles(Select)
    className="MyPRList-buttonDesktop-5"
    displayEmpty={true}
    name="Occasion"
    onChange={[Function]}
    value="ALL"
  >
    <WithStyles(MenuItem)
      value="ALL"
    >
      <div
        className="MyPRList-filterDesktop-6"
      >
        Alle
      </div>
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="ON_DEMAND"
      value="ON_DEMAND"
    >
      <div
        className="MyPRList-filterDesktop-6"
      >
        Auf Nachfrage
      </div>
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="YEARLY"
      value="YEARLY"
    >
      <div
        className="MyPRList-filterDesktop-6"
      >
        jährlich
      </div>
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="QUARTERLY"
      value="QUARTERLY"
    >
      <div
        className="MyPRList-filterDesktop-6"
      >
        vierteljährlich
      </div>
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="END_PROBATION"
      value="END_PROBATION"
    >
      <div
        className="MyPRList-filterDesktop-6"
      >
        Ende der Probezeit
      </div>
    </WithStyles(MenuItem)>
  </WithStyles(Select)>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(Button)
      className="MyPRList-buttonDesktop-5"
      onClick={[Function]}
      variant="raised"
    >
      PR beantragen
      <WithStyles(Icon)>
        add
      </WithStyles(Icon)>
    </WithStyles(Button)>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <div
      className="MyPRList-container-1"
    >
      <div
        className="MyPRList-root-2"
      >
        <WithStyles(Button)
          className="MyPRList-buttonList-19"
          key="1"
          onClick={[Function]}
        >
          <WithStyles(List)
            className="MyPRList-openList-8"
          >
            <WithStyles(MyPRListItem)
              deadline="2015-05-11"
              occasion="jährlich"
              reviewer="ttran"
              status="In Vorbereitung"
            />
          </WithStyles(List)>
        </WithStyles(Button)>
        <WithStyles(Button)
          className="MyPRList-buttonList-19"
          key="2"
          onClick={[Function]}
        >
          <WithStyles(List)
            className="MyPRList-list-7"
          >
            <WithStyles(MyPRListItem)
              deadline="2222-05-11"
              occasion="Auf Nachfrage"
              reviewer="test"
              status="In Vorbereitung"
            />
          </WithStyles(List)>
        </WithStyles(Button)>
      </div>
      <div
        className="MyPRList-cardContainerRow-16"
      >
        <WithStyles(Card)
          className="MyPRList-cardColumnSheet-17"
        >
          <WithStyles(Typography)
            className="MyPRList-title-15"
            variant="body2"
          >
            SHEET
          </WithStyles(Typography)>
          <Connect(WithStyles(PrSheet))
            key="1"
            prById={
              Object {
                "_links": Object {
                  "self": Object {
                    "href": "http://localhost:8010/api/v1/prs/1",
                  },
                },
                "deadline": "2015-05-11",
                "employee": Object {
                  "email": "lionel.schäfer@senacor.com",
                  "employment": Object {
                    "endOfProbationPeriod": "2004-05-10",
                    "jobGradings": Array [
                      Object {
                        "competence": "DEVELOPMENT",
                        "positionId": 1,
                        "salaryLevel": 1,
                        "validFrom": "2004-05-09",
                      },
                      Object {
                        "competence": "DEVELOPMENT",
                        "positionId": 2,
                        "salaryLevel": 2,
                        "validFrom": "2005-05-09",
                      },
                    ],
                    "leaves": Object {
                      "maternityLeave": Array [
                        Object {
                          "from": "2004-05-10",
                          "to": "2004-10-11",
                        },
                      ],
                      "parentalLeave": Array [
                        Object {
                          "from": "2004-05-10",
                          "to": "2004-08-28",
                        },
                      ],
                      "sabbatical": Array [
                        Object {
                          "from": "2004-05-10",
                          "to": "2004-06-07",
                        },
                        Object {
                          "from": "2004-05-10",
                          "to": "2005-02-10",
                        },
                        Object {
                          "from": "2004-05-10",
                          "to": "2005-03-26",
                        },
                      ],
                      "unpaidLeave": Array [
                        Object {
                          "from": "2004-05-10",
                          "to": "2005-03-31",
                        },
                      ],
                    },
                  },
                  "entryDate": "2004-05-10",
                  "firstName": "Lionel",
                  "id": 1,
                  "lastName": "Schäfer",
                  "login": "lschäfer",
                  "salaries": Array [
                    Object {
                      "basicSalary": 40500,
                      "fte": 1,
                      "ote": 45000,
                      "validFrom": "2004-05-10",
                      "workingHoursPerWeek": 40,
                    },
                    Object {
                      "basicSalary": 40000,
                      "fte": 1,
                      "ote": 50000,
                      "validFrom": "2005-05-10",
                      "workingHoursPerWeek": 40,
                    },
                  ],
                  "title": "DR",
                },
                "id": 1,
                "occasion": "YEARLY",
                "status": "PREPARATION",
                "supervisor": "ttran",
              }
            }
          />
        </WithStyles(Card)>
        <div
          className="MyPRList-cardContainerColumn-18"
        >
          <WithStyles(Card)
            className="MyPRList-cardColumn-14"
          >
            <WithStyles(Typography)
              className="MyPRList-title-15"
              variant="body2"
            >
              GEHALT UND ANSTELLUNG
            </WithStyles(Typography)>
            <WithStyles(PrSalary)
              prById={
                Object {
                  "_links": Object {
                    "self": Object {
                      "href": "http://localhost:8010/api/v1/prs/1",
                    },
                  },
                  "deadline": "2015-05-11",
                  "employee": Object {
                    "email": "lionel.schäfer@senacor.com",
                    "employment": Object {
                      "endOfProbationPeriod": "2004-05-10",
                      "jobGradings": Array [
                        Object {
                          "competence": "DEVELOPMENT",
                          "positionId": 1,
                          "salaryLevel": 1,
                          "validFrom": "2004-05-09",
                        },
                        Object {
                          "competence": "DEVELOPMENT",
                          "positionId": 2,
                          "salaryLevel": 2,
                          "validFrom": "2005-05-09",
                        },
                      ],
                      "leaves": Object {
                        "maternityLeave": Array [
                          Object {
                            "from": "2004-05-10",
                            "to": "2004-10-11",
                          },
                        ],
                        "parentalLeave": Array [
                          Object {
                            "from": "2004-05-10",
                            "to": "2004-08-28",
                          },
                        ],
                        "sabbatical": Array [
                          Object {
                            "from": "2004-05-10",
                            "to": "2004-06-07",
                          },
                          Object {
                            "from": "2004-05-10",
                            "to": "2005-02-10",
                          },
                          Object {
                            "from": "2004-05-10",
                            "to": "2005-03-26",
                          },
                        ],
                        "unpaidLeave": Array [
                          Object {
                            "from": "2004-05-10",
                            "to": "2005-03-31",
                          },
                        ],
                      },
                    },
                    "entryDate": "2004-05-10",
                    "firstName": "Lionel",
                    "id": 1,
                    "lastName": "Schäfer",
                    "login": "lschäfer",
                    "salaries": Array [
                      Object {
                        "basicSalary": 40500,
                        "fte": 1,
                        "ote": 45000,
                        "validFrom": "2004-05-10",
                        "workingHoursPerWeek": 40,
                      },
                      Object {
                        "basicSalary": 40000,
                        "fte": 1,
                        "ote": 50000,
                        "validFrom": "2005-05-10",
                        "workingHoursPerWeek": 40,
                      },
                    ],
                    "title": "DR",
                  },
                  "id": 1,
                  "occasion": "YEARLY",
                  "status": "PREPARATION",
                  "supervisor": "ttran",
                }
              }
            />
            <WithStyles(PrEmployment)
              prById={
                Object {
                  "_links": Object {
                    "self": Object {
                      "href": "http://localhost:8010/api/v1/prs/1",
                    },
                  },
                  "deadline": "2015-05-11",
                  "employee": Object {
                    "email": "lionel.schäfer@senacor.com",
                    "employment": Object {
                      "endOfProbationPeriod": "2004-05-10",
                      "jobGradings": Array [
                        Object {
                          "competence": "DEVELOPMENT",
                          "positionId": 1,
                          "salaryLevel": 1,
                          "validFrom": "2004-05-09",
                        },
                        Object {
                          "competence": "DEVELOPMENT",
                          "positionId": 2,
                          "salaryLevel": 2,
                          "validFrom": "2005-05-09",
                        },
                      ],
                      "leaves": Object {
                        "maternityLeave": Array [
                          Object {
                            "from": "2004-05-10",
                            "to": "2004-10-11",
                          },
                        ],
                        "parentalLeave": Array [
                          Object {
                            "from": "2004-05-10",
                            "to": "2004-08-28",
                          },
                        ],
                        "sabbatical": Array [
                          Object {
                            "from": "2004-05-10",
                            "to": "2004-06-07",
                          },
                          Object {
                            "from": "2004-05-10",
                            "to": "2005-02-10",
                          },
                          Object {
                            "from": "2004-05-10",
                            "to": "2005-03-26",
                          },
                        ],
                        "unpaidLeave": Array [
                          Object {
                            "from": "2004-05-10",
                            "to": "2005-03-31",
                          },
                        ],
                      },
                    },
                    "entryDate": "2004-05-10",
                    "firstName": "Lionel",
                    "id": 1,
                    "lastName": "Schäfer",
                    "login": "lschäfer",
                    "salaries": Array [
                      Object {
                        "basicSalary": 40500,
                        "fte": 1,
                        "ote": 45000,
                        "validFrom": "2004-05-10",
                        "workingHoursPerWeek": 40,
                      },
                      Object {
                        "basicSalary": 40000,
                        "fte": 1,
                        "ote": 50000,
                        "validFrom": "2005-05-10",
                        "workingHoursPerWeek": 40,
                      },
                    ],
                    "title": "DR",
                  },
                  "id": 1,
                  "occasion": "YEARLY",
                  "status": "PREPARATION",
                  "supervisor": "ttran",
                }
              }
            />
          </WithStyles(Card)>
          <WithStyles(Card)
            className="MyPRList-cardColumn-14"
          >
            <WithStyles(Typography)
              className="MyPRList-title-15"
              variant="body2"
            >
              STATUS
            </WithStyles(Typography)>
            <WithStyles(PrState) />
          </WithStyles(Card)>
        </div>
      </div>
    </div>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={false}
    smUp={true}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <Link
      key="1"
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/prs/1"
    >
      <WithStyles(List)
        className="MyPRList-list-7"
      >
        <WithStyles(MyPRListItem)
          deadline="2015-05-11"
          occasion="jährlich"
          reviewer="ttran"
          status="In Vorbereitung"
        />
      </WithStyles(List)>
    </Link>
    <Link
      key="2"
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/prs/2"
    >
      <WithStyles(List)
        className="MyPRList-list-7"
      >
        <WithStyles(MyPRListItem)
          deadline="2222-05-11"
          occasion="Auf Nachfrage"
          reviewer="test"
          status="In Vorbereitung"
        />
      </WithStyles(List)>
    </Link>
  </Hidden>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={false}
    smUp={true}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(Button)
      aria-label="add"
      className="MyPRList-buttonMobile-3"
      color="primary"
      onClick={[Function]}
      variant="fab"
    >
      <pure(Add)
        className="MyPRList-icon-4"
      />
    </WithStyles(Button)>
  </Hidden>
</div>
`;
